### PR TITLE
Add some additional dsi functions revealed by the arrays/userAPI test

### PR DESCRIPTION
**Pull request by lydia-duncan**
Fix the param argument version of dsiDim, adds dsiHigh, dsiAlignedLow,
dsiAlignedHigh, dsiSerialReadWrite, dsiSerialRead, and dsiSerialWrite to the
domain, adds dsiLocalAccess, dsiTargetLocales, dsiHasSingleLocalSubdomain,
dsiLocalSubdomain, and dsiLocalSubdomains to the array.

Note that the userAPI test cannot currently work with ExternArr because the
test interface assumes at least two dimensions and we only support one dimension
today.

Full paratest with futures in progress

### CODE REVIEW

**Comment by mppf** [comment]
@lydia-duncan - I'm presuming you found these issues with a modified version of the userAPI test. If so, can we commit the 1-D version that tests this distribution?

**Comment by lydia-duncan** [comment]
@bradcray  was skeptical that it was worth taking the effort now, but I may have misunderstood him (it did seem odd that we wouldn't want to test it, but maybe the rationale was that the implementation might change soon anyways?)

**Comment by lydia-duncan** [comment]
I do have a partial version of the test stored away on a branch

**Comment by mppf** [comment]
@lydia-duncan - it's unclear to me how much effort it is. If it's 1 hour (say) I'd think it'd be worth it.

File: modules/internal/ExternalArray.chpl [hunk 1]
```diff
@@ -165,7 +165,7 @@ module ExternalArray {
       if (d != rank) {
```
Comment (mppf) [inline modules/internal/ExternalArray.chpl]: Remove the "Necessary?" comment above?

**Comment by lydia-duncan** [comment]
I'm running into issues with the leader/follower iterators.  They seem to be generating too many iterations, or at least more than an equivalent DefaultRectangular was.  Pausing for now to look at other things

**Pull request closed without merging**